testbed/dataconnectors/routing.go

```diff
@@ -0,0 +1,36 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package dataconnectors // import "github.com/open-telemetry/opentelemetry-collector-contrib/testbed/dataconnectors"
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/testbed"
+)
+
+type RoutingDataConnector struct {
+	testbed.DataConnectorBase
+}
+
+var _ testbed.DataConnector = (*RoutingDataConnector)(nil)
+
+func NewRoutingDataConnector(receiverDataType string) *RoutingDataConnector {
+	return &RoutingDataConnector{DataConnectorBase: testbed.DataConnectorBase{ReceiverDataType: receiverDataType}}
+}
+
+func (rc *RoutingDataConnector) GenConfigYAMLStr() string {
+	// Note that this generates an exporter config for agent.
+	return `
+  routing:
+    table:
+      - statement: route()
+        pipelines: [traces/out]`
```

Comment:
Do you plan to iterate on this in a future PR? I get that you're mostly just adding the framework now but this particular test doesn't appear to validate much.

Reply:
Yes we could iterate on this in a future PR but just wanted to add this test to validate that connectors are included within the config and the connector is working as expected.